mavlink-core/src/bytes.rs

```
@@ -40,14 +40,10 @@ impl<'a> Bytes<'a> {
     }
 
     #[inline]
-    fn check_remaining(&self, count: usize) -> Result<(), Error> {
-        if self.remaining() >= count {
-            Ok(())
-        } else {
-            Err(Error::NotEnoughBuffer {
-                requested: count,
-                available: self.remaining(),
-            })
+    fn bounds_error(&self, requested: usize) -> Error {
+        Error::NotEnoughBuffer {
+            requested,
+            available: self.remaining(),
```

You can implement `fn not_enough_buffer` under `impl Error` directly.

---

Could you maybe clarify ? I can't move `fn bounds_error` itself since it has to call `self.remaining()` and creating a function just to wrap `Error::NotEnoughBuffer { requested, available}` seems like unnecesarry bloat since it would do nothing extra and only be used in this one place.

---

You can pass the remaining value the same way you pass the requested value. The bloating is already happening with the current approach and it's not just bloat but also an uncommon way to return errors. If you want to avoid bloat, you should return the error directly instead of having to call a function.